Node/samples/app.js

```diff
@@ -130,7 +134,187 @@ bot.dialog('RouteMessageToGeneral', function (session) {
+            .includeTime(false)
+            .isRequired(false)
+    ])
+        .actions([
+        new teams.O365ConnectorCardHttpPOST(session)
+            .id("card-3-btn-1")
+            .name("Send")
+            .body(JSON.stringify({
+            date1: '{{date-1.value}}',
+            date2: '{{date-2.value}}'
+        }))
+    ]);
+    var section = new teams.O365ConnectorCardSection(session)
+        .markdown(true)
+        .title("**section title**")
+        .text("section text")
+        .activityTitle("activity title")
+        .activitySubtitle("activity sbtitle")
+        .activityImage("http://connectorsdemo.azurewebsites.net/images/MSC12_Oscar_002.jpg")
+        .activityText("activity text")
+        .facts([
+        new teams.O365ConnectorCardFact(session).name("Fact Name 1").value("Faction value 1"),
+        new teams.O365ConnectorCardFact(session).name("Fact Name 2").value("Faction value 2"),
+    ])
+        .images([
+        new teams.O365ConnectorCardImage(session).title("image 1").image("http://connectorsdemo.azurewebsites.net/images/MicrosoftSurface_024_Cafe_OH-06315_VS_R1c.jpg"),
+        new teams.O365ConnectorCardImage(session).title("image 2").image("http://connectorsdemo.azurewebsites.net/images/WIN12_Scene_01.jpg"),
+        new teams.O365ConnectorCardImage(session).title("image 3").image("http://connectorsdemo.azurewebsites.net/images/WIN12_Anthony_02.jpg")
+    ]);
+    var card = new teams.O365ConnectorCard(session)
+        .summary("O365 card summar")
+        .themeColor("#8899AA")
+        .title("card title")
+        .text("card text")
+        .sections([section])
+        .potentialAction([
+        actionCard1,
+        actionCard2,
+        actionCard3,
+        new teams.O365ConnectorCardViewAction(session)
+            .name('View Action')
+            .target('http://microsoft.com'),
+        new teams.O365ConnectorCardOpenUri(session)
+            .id('open-uri')
+            .name('Open Uri')["default"]('http://microsoft.com')
+            .iOS('http://microsoft.com')
+            .android('http://microsoft.com')
+            .windowsPhone('http://microsoft.com')
+    ]);
+    var msg = new teams.TeamsMessage(session)
+        .summary("A sample O365 actionable card")
+        .attachments([card]);
+    session.send(msg);
+    session.endDialog();
 });
+// example for o365 connector actionable card
+var o365CardActionHandler = function (event, query, callback) {
+    var userName = event.address.user.name;
+    var body = JSON.parse(query.body);
+    var msg = new builder.Message()
+        .address(event.address)
+        .summary("Thanks for your input!")
+        .textFormat("xml")
+        .text("<h2>Thanks, " + userName + "!<br/>\n                   <h3>Your input action ID:</h3><br/>\n                   <pre>" + query.actionId + "</pre><br/>\n                   <h3>Your input body:</h3><br/>\n                   <pre>" + JSON.stringify(body, null, 2) + "</pre>");
```

Comment:
It's to send an update response message in HTML format. It's in multiple lines. This is compiled results. Please check out app.ts for original format.